api/src/main/java/org/hawkular/inventory/api/Relationships.java

```diff
@@ -91,14 +117,17 @@ private Relationships() {
     /**
      * Provides read-write access to relationships.
      */
-    public interface ReadWrite extends ReadWriteInterface<Relationship, Relationship.Blueprint, Single, Multiple> {
```

Comment:
I see, I agree it is quite weird. It's actually similar for the `Multiple` iface, but it makes more sense there as the .tenants()/.environments()/.resources()/.etcs() methods works as an additional filter. Basically the same way as the `TargetOfType` filter now works (provided the outgoing relationships were chosen).

For the Single it's the same, it's just an additional filter on a type, I admit, it doesn't make much sense for the Single, perhaps something like `getMeTheOtherEndOfThisRelationshipPlease()` makes more sense here. With the cost of losing the type safety, or perhaps this would be solvable as well with some lookahead to the "target"'s type in the hawk-pipe.